main.tex

```diff
@@ -225,47 +497,162 @@ \subsubsection{On-ChainCircuit Breaker and the Breaker Box}
+  \item \textbf{Redemption fee}\,:  The fee is paid in collateral and added \emph{directly} to the redeemed trove rather than to veMENTO stakers.  This compensates the borrower for the forced deleveraging and preserves peg stability.
 \end{itemize}
 
-\subsubsection{Challenges}
+\subsubsection*{FPMM Liquidity Provider Incentives}
 \begin{itemize}
-    \item Fragmented Liquidity: The ease and affordability of launching new decentralized stablecoins pose challenges, such as the potential for multiple parties to introduce multiple versions of the same asset, leading to fragmented liquidity. To mitigate this, Mento will implement an incentivization module, enabling operators and users to incentivize borrowers and liquidity providers. Through the Gauge System, holders of MENTO governance tokens can direct a portion of future token emissions to specific asset stability pools or liquidity pools, fostering liquidity for meaningful assets while rendering less favored 
-    assets irrelevant.
-    \item Safeguarding Users: Ensuring end-user protection against stablecoins launched with questionable economic setups or malicious intent is paramount. While operators have the flexibility to select any collateral asset, the requirement of a price feed from data providers like RedStone limits the assets eligible for use as collateral. Additionally, the Gauge System acts as a second line of defense, allowing the community to review and reject assets lacking sound economic fundamentals through governance decisions.
-    \item Governance and Revenue Sharing: The fee income generated by the Stablecoin Factory is expected to be allocated between the Mento treasury, Mento contributors and MENTO token holders participating in the governance process. Token holders can vote on specific stablecoin pools, ensuring community involvement in decision-making processes.
+  \item \textbf{Spread capture}\,:  FPMM LPs quote a fixed bid/ask around the oracle price and earn the embedded spread on each user swap.
+  \item \textbf{MENTO emissions via gauges}\,:  A governor\hyp controlled ``gauge'' streams a capped USD value of MENTO tokens to staked LP tokens, ensuring competitive on\hyp chain liquidity without overpaying.
+  \item \textbf{Swap fee (\texttt{amm\_fee})}\,:  Every trade pays an additional micro\hyp fee that flows \emph{entirely} to LPs during normal, two\hyp sided volume.
+\end{itemize}
+
+\subsubsection*{Keeper Bots for Flash\hyp Swaps}
+\begin{itemize}
+\item A network of permissionless keeper bots monitors inventory drift.  When the pool breaches its tolerance band they atomically call \verb|rebalance(poolId)|, executing the flash\hyp swap.  Each successful call earns a flat, governor\hyp set MENTO bounty (denominated in USD) sized at roughly $1.5\times$ the prevailing gas cost so that maintenance remains profitable on any chain.
```

Comment:
i believe currently the code actually has a rebalancing reward that is paid out in either token A or token B of an fpmm, maybe I'm missing something but whitepaper and code don't seem aligned here

/cc @baroooo @philbow61

Reply:
@baroooo @philbow61 can you let me know how this is currently implemented?